pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -170,19 +171,31 @@ func (r *ReconcileMasterUserRecord) ensureUserAccount(logger logr.Logger, murAcc
 			// does not exist - should create
 			userAccount = newUserAccount(nsdName, murAccount.Spec, mur.Spec)
 			if err := memberCluster.Client.Create(context.TODO(), userAccount); err != nil {
-				return false, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToCreateUserAccountReason), err,
+				return 0, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToCreateUserAccountReason), err,
 					"failed to create UserAccount in the member cluster '%s'", murAccount.TargetCluster)
 			}
-			return false, updateStatusConditions(logger, r.client, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, ""))
+			counter.IncrementUserAccountCount(murAccount.TargetCluster)
+			return 0, updateStatusConditions(logger, r.client, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, ""))
 		}
 		// another/unexpected error occurred while trying to fetch the user account on the member cluster
-		return false, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToGetUserAccountReason), err,
+		return 0, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToGetUserAccountReason), err,
 			"failed to get userAccount '%s' from cluster '%s'", mur.Name, murAccount.TargetCluster)
 	}
 	// if the UserAccount is being deleted (by accident?), then we should wait until is has been totally deleted, and this controller will recreate it again
 	if util.IsBeingDeleted(userAccount) {
 		logger.Info("UserAccount is being deleted. Waiting until deletion is complete", "member_cluster", memberCluster.Name)
-		return true, updateStatusConditions(logger, r.client, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "recovering deleted UserAccount"))
+		deletionTimestamp := userAccount.GetDeletionTimestamp()
+
+		requeueTime := 3 * time.Second
+		timeUntilDeletion := time.Until(deletionTimestamp.Time)
+		if timeUntilDeletion+3*time.Second >= 0 {
+			counter.DecrementUserAccountCount(logger, murAccount.TargetCluster)
```

Comment:
ok, so if I understand correctly, the code here makes sure that we will requeue but only 3s after the deletion timestamp, but we should not update the counter twice? 
Maybe you could add a comment to explain the reasoning? (for later, when we come back to this code)

Reply:
Yes, you are right with the understanding. 
Ok, I'll add the explanation there - maybe I'll copy-paste your comment :-)

Reply:
sure, feel free to reuse my prose! 😉

Reply:
done 7623326